src/utils/array.js

```diff
@@ -0,0 +1,27 @@
+// @flow
+/*
+    Pillar Wallet: the personal data locker
+    Copyright (C) 2021 Stiftung Pillar Project
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+/**
+ * Performs map operation and filters out `null` and `undefined` results.
+ */
+export function mapNotNil<Item, Result>(items: Item[], selector: (Item) => ?Result): Result[] {
+  // $FlowFixMe: does not infer that `filter` removes nil values
+  return items.map((item) => selector(item)).filter((item) => item != null);
```

Comment:
`selector` type implies that the function also filters out `undefined` -- which it does, but because it uses `!=` instead of `!==`. I'd rather not rely on that kind of detail about the language, though.

Also, since this is a wrapper for `map`, it could support all arguments that are usually passed to selector.

In general, good idea, since I've seen it repeated a few times.

Reply:
Re naming: I used `NotNil` because popular libs like lodash, ramda, etc use `nil` to signify `null` or `undefined` (https://lodash.com/docs/4.17.15#isNil)

Reply:
Re `!= null`: I consider this to be standard JS idiom meaning `!== null & !== undefined` on par with `!` and `!!` (falsy handling).

Reply:
Re selector args: I'll add these to match `map`

Reply:
No, nil is fine, though now that you pointed it out the whole name sounds a bit like filter is applied first. Re `!=` I must not have seen it used enough to see it as a standard 🙂 but it does look useful as long as one knows what one's doing.